samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -483,24 +490,14 @@ object SamzaContainer extends Logging {
 
       val taskName = taskModel.getTaskName
 
-      val taskObj = if (taskFactory != null) {
-        debug("Using task factory to create task instance")
-        taskFactory match {
+      debug("Using task factory to create task instance")
+      val task =
+        finalTaskFactory match {
```

Comment:
Do we still need this match statement since we are already validating taskFactory instance in TaskFactories?